### PR TITLE
Update MergeEvent changes struct

### PR DESCRIPTION
**Pull request by nauxliu**

### CODE REVIEW

File: event_types.go [hunk 1]
```diff
@@ -579,10 +579,11 @@ type MergeEvent struct {
...
 	Changes struct {
-		AssigneeID struct {
-			Previous int `json:"previous"`
-			Current  int `json:"current"`
-		} `json:"assignee_id"`
+		Assignees []struct {
+			Name      string `json:"name"`
+			Username  string `json:"username"`
+			AvatarURL string `json:"avatar_url"`
+		} `json:"assignees"`
```
Comment (svanharmelen) [inline event_types.go]: I don't see these changes in the docs. Could you point me towards a location (in the docs or the source code of Gitlab) or add some API responses that show that this is in fact changed? Thanks!

**Comment by nauxliu** [comment]
@svanharmelen 

Here is a merge request webhook request:

```json
{
    "object_kind":"merge_request",
    "event_type":"merge_request",
    "user":{
        "name":"Xiangxuan Liu",
        "username":"NauxLiu",
        "avatar_url":"https://assets.gitlab-static.net/uploads/-/system/user/avatar/363406/avatar.png"
    },
    "project":{
        "id":4184318,
        "name":"test",
        "description":"",
        "web_url":"https://gitlab.com/NauxLiu/test",
        "avatar_url":null,
        "git_ssh_url":"git@gitlab.com:NauxLiu/test.git",
        "git_http_url":"https://gitlab.com/NauxLiu/test.git",
        "namespace":"Xiangxuan Liu",
        "visibility_level":0,
        "path_with_namespace":"NauxLiu/test",
        "default_branch":"master",
        "ci_config_path":null,
        "homepage":"https://gitlab.com/NauxLiu/test",
        "url":"git@gitlab.com:NauxLiu/test.git",
        "ssh_url":"git@gitlab.com:NauxLiu/test.git",
        "http_url":"https://gitlab.com/NauxLiu/test.git"
    },
    "object_attributes":{
        "assignee_id":363406,
        "author_id":363406,
        "created_at":"2019-10-23 06:37:28 UTC",
        "description":"foobar",
        "head_pipeline_id":null,
        "id":40352048,
        "iid":1,
        "last_edited_at":null,
        "last_edited_by_id":null,
        "merge_commit_sha":null,
        "merge_error":null,
        "merge_params":{
            "force_remove_source_branch":"1"
        },
        "merge_status":"unchecked",
        "merge_user_id":null,
        "merge_when_pipeline_succeeds":false,
        "milestone_id":null,
        "source_branch":"test",
        "source_project_id":4184318,
        "state":"opened",
        "target_branch":"master",
        "target_project_id":4184318,
        "time_estimate":0,
        "title":"WIP: Test",
        "updated_at":"2019-10-23 06:37:28 UTC",
        "updated_by_id":null,
        "url":"https://gitlab.com/NauxLiu/test/merge_requests/1",
        "source":{
            "id":4184318,
            "name":"test",
            "description":"",
            "web_url":"https://gitlab.com/NauxLiu/test",
            "avatar_url":null,
            "git_ssh_url":"git@gitlab.com:NauxLiu/test.git",
            "git_http_url":"https://gitlab.com/NauxLiu/test.git",
            "namespace":"Xiangxuan Liu",
            "visibility_level":0,
            "path_with_namespace":"NauxLiu/test",
            "default_branch":"master",
            "ci_config_path":null,
            "homepage":"https://gitlab.com/NauxLiu/test",
            "url":"git@gitlab.com:NauxLiu/test.git",
            "ssh_url":"git@gitlab.com:NauxLiu/test.git",
            "http_url":"https://gitlab.com/NauxLiu/test.git"
        },
        "target":{
            "id":4184318,
            "name":"test",
            "description":"",
            "web_url":"https://gitlab.com/NauxLiu/test",
            "avatar_url":null,
            "git_ssh_url":"git@gitlab.com:NauxLiu/test.git",
            "git_http_url":"https://gitlab.com/NauxLiu/test.git",
            "namespace":"Xiangxuan Liu",
            "visibility_level":0,
            "path_with_namespace":"NauxLiu/test",
            "default_branch":"master",
            "ci_config_path":null,
            "homepage":"https://gitlab.com/NauxLiu/test",
            "url":"git@gitlab.com:NauxLiu/test.git",
            "ssh_url":"git@gitlab.com:NauxLiu/test.git",
            "http_url":"https://gitlab.com/NauxLiu/test.git"
        },
        "last_commit":{
            "id":"110bdaed3a35a320bd0ba929239f9323b9cd7d53",
            "message":"Add new file",
            "timestamp":"2017-09-19T10:51:50Z",
            "url":"https://gitlab.com/NauxLiu/test/commit/110bdaed3a35a320bd0ba929239f9323b9cd7d53",
            "author":{
                "name":"刘相轩",
                "email":"nauxliu@gmail.com"
            }
        },
        "work_in_progress":true,
        "total_time_spent":0,
        "human_total_time_spent":null,
        "human_time_estimate":null,
        "assignee_ids":[
            363406
        ],
        "action":"open"
    },
    "labels":[

    ],
    "changes":{
        "author_id":{
            "previous":null,
            "current":363406
        },
        "created_at":{
            "previous":null,
            "current":"2019-10-23 06:37:28 UTC"
        },
        "description":{
            "previous":null,
            "current":"foobar"
        },
        "id":{
            "previous":null,
            "current":40352048
        },
        "iid":{
            "previous":null,
            "current":1
        },
        "merge_params":{
            "previous":{

            },
            "current":{
                "force_remove_source_branch":"1"
            }
        },
        "source_branch":{
            "previous":null,
            "current":"test"
        },
        "source_project_id":{
            "previous":null,
            "current":4184318
        },
        "target_branch":{
            "previous":null,
            "current":"master"
        },
        "target_project_id":{
            "previous":null,
            "current":4184318
        },
        "title":{
            "previous":null,
            "current":"WIP: Test"
        },
        "updated_at":{
            "previous":null,
            "current":"2019-10-23 06:37:28 UTC"
        },
        "assignees":{
            "previous":[

            ],
            "current":[
                {
                    "name":"Xiangxuan Liu",
                    "username":"NauxLiu",
                    "avatar_url":"https://assets.gitlab-static.net/uploads/-/system/user/avatar/363406/avatar.png"
                }
            ]
        },
        "total_time_spent":{
            "previous":null,
            "current":0
        }
    },
    "repository":{
        "name":"test",
        "url":"git@gitlab.com:NauxLiu/test.git",
        "description":"",
        "homepage":"https://gitlab.com/NauxLiu/test"
    },
    "assignees":[
        {
            "name":"Xiangxuan Liu",
            "username":"NauxLiu",
            "avatar_url":"https://assets.gitlab-static.net/uploads/-/system/user/avatar/363406/avatar.png"
        }
    ]
}
```

**Comment by svanharmelen** [comment]
Thanks @nauxliu. It looks like that response uses a slightly different format as it still contains a previous and current list which you didn't add in your PR:

```
 "assignees":{
            "previous":[

            ],
            "current":[
                {
                    "name":"Xiangxuan Liu",
                    "username":"NauxLiu",
                    "avatar_url":"https://assets.gitlab-static.net/uploads/-/system/user/avatar/363406/avatar.png"
                }
            ]
        },
```

**Comment by nauxliu** [comment]
Oops, my mistake.

**Comment by nauxliu** [comment]
@svanharmelen Thanks for point out my stupid mistake, I've fixed that and added some test to make sure it works.

**Pull request merged**